sdk/core/Azure.Core/api/Azure.Core.net472.cs

```diff
@@ -776,10 +778,15 @@ public readonly partial struct TokenRequestContext
         public TokenRequestContext(string[] scopes, string? parentRequestId) { throw null; }
         public TokenRequestContext(string[] scopes, string? parentRequestId, string? claims) { throw null; }
         public TokenRequestContext(string[] scopes, string? parentRequestId, string? claims, string? tenantId) { throw null; }
-        public TokenRequestContext(string[] scopes, string? parentRequestId = null, string? claims = null, string? tenantId = null, bool isCaeEnabled = false) { throw null; }
+        public TokenRequestContext(string[] scopes, string? parentRequestId, string? claims, string? tenantId, bool isCaeEnabled) { throw null; }
+        public TokenRequestContext(string[] scopes, string? parentRequestId = null, string? claims = null, string? tenantId = null, bool isCaeEnabled = false, bool isProofOfPossessionEnabled = false, string? proofOfPossessionNonce = null, Azure.Core.Request? request = null) { throw null; }
         public string? Claims { get { throw null; } }
         public bool IsCaeEnabled { get { throw null; } }
+        public bool IsProofOfPossessionEnabled { get { throw null; } }
```

Comment:
Is the idea behind this property that if it is true, the nullable properties being added are expected to be populated?

In OO design, often this type of extension is addressed with inheritance so that unused properties don't need to be added to a base type that doesn't use them.  Did we consider such a design and decide against it because structs cannot inherit?  Are there any concerns with adding properties to a value type that will make it larger to copy into calls on the stack?

Reply:
I discussed the tradeoffs here with @KrzysztofCwalina and we considered some alternatives. None seemed to be clearly better. Happy to discuss more offline.